daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Engine.scala

```diff
@@ -411,7 +410,7 @@ class Engine(val config: EngineConfig = Engine.StableConfig) {
 
         case SResultNeedKey(gk, _, cb) =>
           def continueWithCoid = (result: Option[ContractId]) => {
-            discard[Boolean](cb(result))
+            cb(None, result)
```

Comment:
a discarded callback result becomes a callback with `None` for the custom error.